packages/nimble-components/src/table/tests/table-selection.spec.ts

```diff
@@ -648,76 +774,201 @@ describe('Table row selection', () => {
+                        initialSelection: ['1'],
+                        rowToClick: 0,
+                        clickModifiers: { ctrlKey: true },
+                        expectedSelection: ['0', '1'],
+                        fireEvent: true
+                    },
+                    {
+                        name: 'clicking the already selected row maintains its selection and does not emit an event',
+                        initialSelection: ['0'],
+                        rowToClick: 0,
+                        clickModifiers: {},
+                        expectedSelection: ['0'],
+                        fireEvent: false
+                    },
+                    {
+                        name: 'CTRL + clicking the already selected row deselects it and emits an event',
+                        initialSelection: ['0'],
+                        rowToClick: 0,
+                        clickModifiers: { ctrlKey: true },
+                        expectedSelection: [],
+                        fireEvent: true
+                    },
+                    {
+                        name: 'clicking a row with multiple different rows previously selected selects the clicked row and deselects the other row and emits an event',
+                        initialSelection: ['1', '2'],
+                        rowToClick: 0,
+                        clickModifiers: {},
+                        expectedSelection: ['0'],
+                        fireEvent: true
+                    },
+                    {
+                        name: 'CTRL + clicking a row with multiple different rows previously adds the clicked row to the selection and emits an event',
+                        initialSelection: ['1', '2'],
+                        rowToClick: 0,
+                        clickModifiers: { ctrlKey: true },
+                        expectedSelection: ['0', '1', '2'],
+                        fireEvent: true
+                    },
+                    {
+                        name: 'clicking a selected row with multiple rows selected keeps the clicked row selected and deselects others emits an event',
+                        initialSelection: ['1', '2'],
+                        rowToClick: 1,
+                        clickModifiers: {},
+                        expectedSelection: ['1'],
+                        fireEvent: true
+                    },
+                    {
+                        name: 'CTRL + clicking a selected row with multiple rows selected deselects the clicked row, keeps others selected, and emits an event',
+                        initialSelection: ['1', '2'],
+                        rowToClick: 1,
+                        clickModifiers: { ctrlKey: true },
+                        expectedSelection: ['2'],
+                        fireEvent: true
+                    }
+                ];
```

Comment:
Are there `shift` click scenarios that make sense to be part of this set?

Reply:
The `shift` click scenarios are a little more involved, which is why I decided to write specific tests for each scenario I cared about. Technically, I could change this shared test code to be a more flexible and add some of the `shift` click tests to this set, but I decided to keep the shared code simpler.

Reply:
I think I'll group all the shift+click tests together in a `describe` so that it's easier to see all the different scenarios we are testing.